rules/prefer-t-regex.js

```diff
@@ -13,75 +13,134 @@ const create = context => {
 		'falsy'
 	];
 
+	const equalityTests = ['is', 'deepEqual'];
+
 	const findReference = name => {
 		const reference = context.getScope().references.find(reference => reference.identifier.name === name);
 		const definitions = reference.resolved.defs;
+
+		// Many integration tests have identifiers that match zero definitions
+		if (definitions.length === 0) {
+			return undefined;
+		}
+
 		return definitions[definitions.length - 1].node;
 	};
 
+	const isRegExp = lookup => {
+		let isRegex = lookup.regex;
+
+		// It's not a regexp but an identifier
+		if (!isRegex && lookup.type === 'Identifier') {
+			const reference = findReference(lookup.name);
+
+			// Not all possible references have an init field
+			if (reference && reference.init) {
+				isRegex = reference.init.regex;
+			}
+		}
+
+		return isRegex;
+	};
+
+	const booleanHandler = node => {
+		const firstArg = node.arguments[0];
+
+		// First argument is a call expression
+		const isFunctionCall = firstArg.type === 'CallExpression';
+		if (!isFunctionCall || !firstArg.callee.property) {
+			return;
+		}
+
+		const {name} = firstArg.callee.property;
+		let lookup = {};
+		let variable = {};
+
+		if (name === 'test') {
+			// `lookup.test(variable)`
+			lookup = firstArg.callee.object;
+			variable = firstArg.arguments[0];
+		} else if (['search', 'match'].includes(name)) {
+			// `variable.match(lookup)`
+			lookup = firstArg.arguments[0];
+			variable = firstArg.callee.object;
+		}
+
+		if (!isRegExp(lookup)) {
+			return;
+		}
+
+		const assertion = ['true', 'truthy'].includes(node.callee.property.name) ? 'regex' : 'notRegex';
+
+		const fix = fixer => {
+			const source = context.getSourceCode();
+			return [
+				fixer.replaceText(node.callee.property, assertion),
+				fixer.replaceText(firstArg, `${source.getText(variable)}, ${source.getText(lookup)}`)
+			];
+		};
+
+		context.report({
+			node,
+			message: `Prefer using the \`t.${assertion}()\` assertion.`,
+			fix
+		});
+	};
+
+	const equalityHandler = node => {
+		const firstArg = node.arguments[0];
+		const secondArg = node.arguments[1];
+
+		const firstIsRx = isRegExp(firstArg);
```

Comment:
Don't use abbreviations.

```suggestion
		const firstArgumentIsRegex = isRegExp(firstArgument);
```

Reply:
Fixed.